pkg/status/combinedstatus-resolution.go

```diff
@@ -475,11 +480,11 @@ func evaluateWorkStatusAgainstStatusCollectorWriteLocked(celEvaluator *celEvalua
 
 	// evaluate groupBy
 	groupByEvals := make(map[string]ref.Val)
-	for _, groupByNamedExp := range scData.GroupBy {
+	for _, groupByNamedExp := range scData.collectorSpec.GroupBy {
 		eval, err := celEvaluator.Evaluate(groupByNamedExp.Def, content)
 		if err != nil {
-			runtime2.HandleError(fmt.Errorf("failed to evaluate groupBy expression: %w", err))
-			continue
+			evalErrors = append(evalErrors, v1alpha1.ErrorInColumn{ColumnName: groupByNamedExp.Name, Error: err.Error()})
+			eval = celtypes.DefaultTypeAdapter.NativeToValue(err.Error())
```

Comment:
Would this duplicate the error both as a row value and a reported error?

Reply:
This is using the reported error string as a cell value in the materialized view being grouped. In other words, it uses the error string as one of the bases for grouping. The code has to do _something_ in this case, and this seems like a reasonable choice.